diffprivlib/SimpleImputer

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+
+class DifferentiallyPrivateImputer(BaseEstimator, TransformerMixin):
+    def __init__(self, missing_values=np.nan, strategy='mean', fill_value=None, epsilon=1.0):
+        self.missing_values = missing_values
+        self.strategy = strategy
+        self.fill_value = fill_value
+        self.epsilon = epsilon  # Privacy budget
+        
+    def fit(self, X, y=None):
+        if self.strategy == 'most_frequent':
+            self.statistics_ = [np.nan] * X.shape[1]  # Not used in most_frequent strategy
+        elif self.strategy == 'constant':
+            self.statistics_ = [self.fill_value] * X.shape[1]
+        else:
+            self.statistics_ = [np.nanmean(col) if np.issubdtype(col.dtype, np.number) else np.nan for col in X.T]
+        return self
+    
+    def _impute_mean(self, col, missing_col):
+        non_missing_values = col[~missing_col]
+        col_mean = np.nanmean(non_missing_values)
+        sensitivity = np.nanmax(np.abs(non_missing_values - col_mean))
+        return col_mean, sensitivity
+    
+    def _impute_median(self, col, missing_col):
+        non_missing_values = col[~missing_col]
+        col_median = np.nanmedian(non_missing_values)
+        sensitivity = np.nanmax(np.abs(non_missing_values - col_median))
```

Comment:
you are using nanmax from numpy... shouldn't you use the dp version from this library ;)